### PR TITLE
Add mocsy

### PR DESCRIPTION
**Pull request by ocefpaf**
@jamesorr and @emiliom I gave up trying to set `mocsy` up as a Python package and surrendered to a hacky conda package that calls `make` and copy the library to the appropriated directory. I still believe we should write wrappers around `mocsy` to make it more user friendly and better `docstring` that Python users are used to, but those do no necessarily need to live in the `mocsy` repo and can be based on this package.

### CODE REVIEW

File: recipes/mocsy/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+make
+
+rm -rf $SP_DIR/numpy
```
Comment (ocefpaf) [inline recipes/mocsy/build.sh]: @msarahan I noticed that `numpy` made into the package `tarball` so I am manually removing it here. Is that an artifact of the `makefile` used here or a `conda-build` bug?

**Comment by emiliom** [comment]
@ocefpaf, thanks for pushing this again! Even if this is not the ideal solution, if it allows for straightforward updates as the mocsy source FORTRAN library is updated, that's a good step forward. FYI, just a few days ago I was recommending mocsy Python access to a modeler, but wondering where things stood. 

I see that there's already a [mocsy package on the conda-forge channel](https://anaconda.org/conda-forge/mocsy). Great! But only Python 3.x. Is a Python 2.7 package coming? I'm not sure if your comment means 2.7 is not doable yet:
> Another limitation of this approach is that, even though we know that Python 2.7 is possible somehow it fails the `import` test

**Comment by ocefpaf** [comment]
>  I'm not sure if your comment means 2.7 is not doable yet:

The CIs failed during the import test (full log: https://circleci.com/gh/ocefpaf/staged-recipes/320?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

```shell
===== testing package: mocsy-2.3.2-np110py27_0 =====
import: 'mocsy'
Traceback (most recent call last):
  File "/staged-recipes/build_artefacts/test-tmp_dir/run_test.py", line 26, in <module>
    import mocsy
ImportError: No module named mocsy
TESTS FAILED: mocsy-2.3.2-np110py27_0
```

That is probably because of the `.so` copy instead of proper `Extension` build. In fact I am surprised the `Python 3` worked! Not sure how to fix that, but maybe it is a simple things. I will check this later over the weekend.

**Comment by emiliom** [comment]
Thanks!

**Pull request merged**